### ISSUE TITLE
[Bug] ”weight=class_weight“ should be modified for ”pos_weight=class_weight“

### ISSUE THREAD

**Issue by fangxu622** [open-mmlab/mmclassification#434]
loss = F.binary_cross_entropy_with_logits(
        pred, label, weight=class_weight, reduction='none') 

it should be modified for  “loss = F.binary_cross_entropy_with_logits( pred, label, pos_weight=class_weight, reduction='none') 
```
def binary_cross_entropy(pred,
                         label,
                         weight=None,
                         reduction='mean',
                         avg_factor=None,
                         class_weight=None):
    r"""Calculate the binary CrossEntropy loss with logits.

    Args:
        pred (torch.Tensor): The prediction with shape (N, \*).
        label (torch.Tensor): The gt label with shape (N, \*).
        weight (torch.Tensor, optional): Element-wise weight of loss with shape
            (N, ). Defaults to None.
        reduction (str): The method used to reduce the loss.
            Options are "none", "mean" and "sum". If reduction is 'none' , loss
            is same shape as pred and label. Defaults to 'mean'.
        avg_factor (int, optional): Average factor that is used to average
            the loss. Defaults to None.
        class_weight (torch.Tensor, optional): The weight for each class with
            shape (C), C is the number of classes. Default None.

    Returns:
        torch.Tensor: The calculated loss
    """
    assert pred.dim() == label.dim()
    # Ensure that the size of class_weight is consistent with pred and label to
    # avoid automatic boracast,
    if class_weight is not None:
        N = pred.size()[0]
        class_weight = class_weight.repeat(N, 1)
    loss = F.binary_cross_entropy_with_logits(
        pred, label, weight=class_weight, reduction='none')

    # apply weights and do the reduction
    if weight is not None:
        assert weight.dim() == 1
        weight = weight.float()
        if pred.dim() > 1:
            weight = weight.reshape(-1, 1)
    loss = weight_reduce_loss(
        loss, weight=weight, reduction=reduction, avg_factor=avg_factor)
    return loss
```

**Comment by Ezra-Yu** [comment]
the class_weight is the weight of C classes, which is designed to be use in multi-label classificiation.

Do you want to add weight for positive and negative categories？

some discuss refer to [#5776](https://github.com/open-mmlab/mmdetection/pull/5776)

**Comment by fangxu622** [comment]
YES! This is for Multilabel image classification.  
I see your discussion.  I still think the should be "pos_weight=class_weight".   you're right that no change when the label is 0.  
The key is that the pos_weights value should setting >1.  It means higher weight compared to the negative sample. So if we think the loss value is large or small. Then we can use the weight parameter not pos_weight to scale the loss value.

**Comment by mzr1996** [comment]
Hello, in binary cross-entropy, `pos_weight` means "a weight of positive examples", instead of "a weight given to each class".
In regular cross-entropy, PyTorch provides `weight`, and describes it as "a manual rescaling weight given to each class.". Therefore, to keep the same behavior, we define `class_weight` as "class-wise weight", differ from `pos_weight`.

<table>
    <tr>
        <th> </th>
        <th colspan=2>class-wise</th>
    </tr>
    <tr>
        <td> </td>
        <td>hat</td>
        <td>glasses</td>
    </tr>
    <tr>
        <td>positive</td>
        <td>0</td>
        <td>1</td>
    </tr>
    <tr>
        <td>negative</td>
        <td>1</td>
        <td>0</td>
    </tr>
</table>

As for your need, we will add the `pos_weight` parameter in our BCE loss recently.

**Comment by fangxu622** [comment]
YES! This is for Multilabel image classification.
I see your discussion. I still think the should be "pos_weight=class_weight". you're right that no change when the label is 0.
The key is that the pos_weights value should setting >1. It means higher weight compared to the negative sample. So if we think the loss value is large or small. Then we can use the weight parameter not pos_weight to scale the loss value.

**Comment by fangxu622** [comment]
> Hello, in binary cross-entropy, `pos_weight` means "a weight of positive examples", instead of "a weight given to each class".
> In regular cross-entropy, PyTorch provides `weight`, and describes it as "a manual rescaling weight given to each class.". Therefore, to keep the same behavior, we define `class_weight` as "class-wise weight", differ from `pos_weight`.
> 
> class-wise
> hat	glasses
> positive	0	1
> negative	1	0
> As for your need, we will add the `pos_weight` parameter in our BCE loss recently.

YES~

**Comment by mzr1996** [comment]
> YES! This is for Multilabel image classification.
> I see your discussion. I think the should be "pos_weight=class_weight". you're right that no change when the label is 0.
> The key is that the pos_weights value should setting >1. It means higher weight compared to the negative sample. So if we think the loss value is large or small. Then we can use the weight parameter not pos_weight to scale the loss value.
> 
> ![image](https://user-images.githubusercontent.com/16681534/132307517-1f8253fc-890a-42f2-b3dc-82e6f8ccadf4.png)

Got it, I think the aim of `pos_weight` is to handle unbalanced positive-negative sample numbers in each class, and the aim of `class_weight` is to handle the different importance of each class. And they're not in conflict.

**Comment by Ezra-Yu** [comment]
we have added pos_weight in BCE refer to #515
